odevalidator/validator.py

```diff
@@ -229,20 +242,47 @@ def _validate(self, data):
     def validate_queue(self, msg_queue):
         results = []
         msg_list = []
+        msg_count = 1
+        # if header, skip over it
+        if self.has_header:
+            header = msg_queue.get()
+            self.check_headers(header)
+
         while not msg_queue.empty():
             line = msg_queue.get()
-            current_msg = json.loads(line)
+            current_msg = self.parse_record[self.data_type](line)
             msg_list.append(current_msg)
-            serial_id = str(current_msg['metadata']['serialId'])
+            record_id = msg_count
+            msg_count += 1
             field_validations = self._validate(current_msg)
-            results.append(RecordValidationResult(serial_id, field_validations, current_msg))
+            results.append(RecordValidationResult(record_id, field_validations, current_msg))
```

Comment:
serialId is required to easily identify the invalid record after the validation results are printed. It used to be record_id but we found that record_id does not sufficiently identify the offending record in case of ODE data.